lapce-core/src/language.rs

```diff
@@ -425,3 +410,77 @@ fn walk_tree(
         cursor.goto_parent();
     }
 }
+
+// NOTE: These tests exist only when `cargo test` is given certain `--features`
+// values, together with `-p lapce-core`. For example:
+//
+//   cargo test -p lapce-core --features lang-rust,lang-python
+//
+// will not run only `test_cpp_lang`. Or use `--all-features`:
+//
+//   cargo test -p lapce-core --all-features
+//
+// In VS Code, clicking the "Run Test" button added to `mod test` will run the
+// test functions only if RA has been given the required features (i.e. in the
+// workspace settings in .vscode/settings.json).
+//
+// If clicking the "Run Test" buttons attached to the functions, RA will add
+// `--feature lang-rust`, for example, for you to the cargo test command line,
+// in addition to any features in its workspace settings.
+#[cfg(test)]
+mod test {
+    #[test]
+    #[cfg(feature = "lang-rust")]
+    fn test_rust_lang() {
+        use super::LapceLanguage;
+        use std::path::PathBuf;
```

Comment:
These look weird. In this function, you import these types, in the others you don't. It would be better to import at the module level and allow unused code on them, I think.